kolibri/auth/permissions/auth.py

```diff
@@ -43,3 +43,13 @@ def user_can_delete_object(self, user, obj):
         else:
             # for non-Facility Collections, defer to the roles to determine delete permissions
             return super(CollectionSpecificRoleBasedPermissions, self).user_can_update_object(user, obj.parent)
+
+class AnybodyCanCreateIfNoDeviceOwner(BasePermissions):
```

Comment:
Some of the test failures are related to this. Unlike what I said, this should actually inherit from `DenyAll`, in order to have the default for the undefined methods be `return False`.

Some of the other tests may be fixed on master -- please merge or rebase.